### PR TITLE
feat: add assist planner template using Python

### PR DESCRIPTION
**Pull request by frankqianms**
workitem: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/27787948

Also upgrade teams-ai dependencies of all Python templates to **V1.1.0**

AI Agent in Teams

### CODE REVIEW

File: templates/python/custom-copilot-assistant-assistants-api/teamsapp.yml.tpl [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  - uses: teamsApp/create
+    with:
+      # Teams app name
+      name: {{appName}}${{TEAMSFX_ENV}}
+    # Write the information of created resources into environment file for
+    # the specified environment variable(s).
+    writeToEnvironmentFile:
+      teamsAppId: TEAMS_APP_ID
+
+  # Create or reuse an existing Azure Active Directory application for bot.
```
Comment (blackchoey) [inline templates/python/custom-copilot-assistant-assistants-api/teamsapp.yml.tpl:20]: Let's use `Microsoft Entra ID ` here, which is AAD's new name. You can refer other templates for how to update the comments.

File: templates/python/custom-copilot-assistant-assistants-api/teamsapp.yml.tpl [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  - uses: botAadApp/create
```
Comment (blackchoey) [inline templates/python/custom-copilot-assistant-assistants-api/teamsapp.yml.tpl:21]: Use `aadApp/create` to create an AAD app for bot

File: templates/python/custom-copilot-assistant-assistants-api/teamsapp.local.yml.tpl [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    with:
+      # Teams app name
+      name: {{appName}}${{TEAMSFX_ENV}}
+    # Write the information of created resources into environment file for
+    # the specified environment variable(s).
+    writeToEnvironmentFile: 
+      teamsAppId: TEAMS_APP_ID
+
+  # Create or reuse an existing Azure Active Directory application for bot.
+  - uses: botAadApp/create
```
Comment (blackchoey) [inline templates/python/custom-copilot-assistant-assistants-api/teamsapp.local.yml.tpl:18]: Use `aadApp/create` to create an AAD app for bot

File: templates/python/custom-copilot-assistant-assistants-api/teamsapp.yml.tpl [hunk 4]
```diff
@@ -78,18 +82,64 @@ provision:
       # Relative path to this file. This is the path for built zip file.
       appPackagePath: ./appPackage/build/appPackage.${{TEAMSFX_ENV}}.zip
 
-# Triggered when 'teamsfx deploy' is executed
+# Triggered when 'teamsapp deploy' is executed
 deploy:
+  # Run npm command
```
Comment (blackchoey) [inline templates/python/custom-copilot-assistant-assistants-api/teamsapp.yml.tpl:87]: Do we need the npm steps?

File: templates/python/custom-copilot-assistant-assistants-api/teamsapp.yml.tpl [hunk 5]
```diff
@@ -78,18 +82,64 @@ provision:
...
+    name: build app
+    with:
+      args: run build --if-present
   # Deploy your application to Azure App Service using the zip deploy feature.
   # For additional details, refer to https://aka.ms/zip-deploy-to-app-services.
   - uses: azureAppService/zipDeploy
     with:
       # Deploy base folder
-      artifactFolder: src
+      artifactFolder: .
```
Comment (blackchoey) [inline templates/python/custom-copilot-assistant-assistants-api/teamsapp.yml.tpl:101]: I saw the deploy folder is changed, is this expected?

File: templates/python/custom-copilot-assistant-assistants-api/teamsapp.yml.tpl (hunk 4 above)
Comment (frankqianms) [reply templates/python/custom-copilot-assistant-assistants-api/teamsapp.yml.tpl]: no we don't. removed

**Review by blackchoey — APPROVED** [review]

**Review by xzf0587 — APPROVED** [review]

**Pull request merged**